packages/vite-plugin-cloudflare/src/utils.ts

```diff
@@ -0,0 +1,74 @@
+import { invariant } from './shared';
+import type { WorkerOptions } from 'miniflare';
+
+function errorMessage(workerName: string) {
+	return `${workerName} does not match a worker name.`;
+}
+
+export function getWorkerEntrypointNames(
+	workers: Array<Pick<WorkerOptions, 'serviceBindings'> & { name: string }>,
+) {
+	const workerEntrypointNames = Object.fromEntries(
+		workers.map((workerOptions) => [workerOptions.name, new Set<string>()]),
+	);
+
+	for (const worker of workers) {
+		if (worker.serviceBindings === undefined) {
+			continue;
+		}
+
+		for (const value of Object.values(worker.serviceBindings)) {
+			if (
+				typeof value === 'object' &&
+				'name' in value &&
+				typeof value.name === 'string' &&
+				value.entrypoint !== undefined &&
+				value.entrypoint !== 'default'
+			) {
+				const entrypointNames = workerEntrypointNames[value.name];
+				invariant(entrypointNames, errorMessage(value.name));
+
+				entrypointNames.add(value.entrypoint);
+			}
+		}
+	}
+
+	return workerEntrypointNames;
+}
+
+export function getDurableObjectClassNames(
+	workers: Array<Pick<WorkerOptions, 'durableObjects'> & { name: string }>,
+) {
+	const durableObjectClassNames = Object.fromEntries(
+		workers.map((workerOptions) => [workerOptions.name, new Set<string>()]),
+	);
+
+	for (const worker of workers) {
+		if (worker.durableObjects === undefined) {
+			continue;
+		}
+
+		for (const value of Object.values(worker.durableObjects)) {
+			if (typeof value === 'string') {
+				const classNames = durableObjectClassNames[worker.name];
+				invariant(classNames, errorMessage(worker.name));
+
+				classNames.add(value);
+			} else if (typeof value === 'object') {
+				if (value.scriptName) {
+					const classNames = durableObjectClassNames[value.scriptName];
+					invariant(classNames, errorMessage(value.scriptName));
+
+					classNames.add(value.className);
```

Comment:
here, are you taking the `scriptName` and using it as the `className`?

this doesn't seem correct to me... is it? 😮

Reply:
Here `scriptName` is the name of the worker where the DO is defined. So we get a reference to the `Set` of exports for that worker and then add the `className` to it.

Reply:
Maybe it's the naming that's confusing. I could call the variables `entryWorkerEntrypointNames` and `entryDurableObjectClassNames` to make it clearer that they are maps of entry files to exports.

Reply:
yeah by debugging it I can see the logic there... but without context the variable names could be a bit confusing 🤔 

I personally would rename the variable and function and all that's related from a generic `DurableObjectClassNames` to something like `WorkersToDurableObjectClassNamesMap` or something like that, it is much more verbose but I personally feel like it makes things much much clearer here 🤔